### PR TITLE
Add logic for syncing reverse solver result

### PR DESCRIPTION
**Pull request by fridex**
## Related Issues and Dependencies

#1725

## This introduces a breaking change

- [ ] Yes
- [x] No

### CODE REVIEW

File: thoth/storages/sync.py [hunk 1]
```diff
@@ -147,6 +148,57 @@ def sync_solver_documents(
...
 
 
+def sync_revsolver_documents(
+    document_ids: Optional[List[str]] = None,
+    force: bool = False,
+    graceful: bool = False,
+    graph: Optional[GraphDatabase] = None,
+    is_local: bool = False,
+) -> tuple:
+    """Sync solver documents into graph."""
```
Comment (pacospace) [inline thoth/storages/sync.py:158]: `revsolver`

File: thoth/storages/sync.py [hunk 2]
```diff
@@ -147,6 +148,57 @@ def sync_solver_documents(
...
+                synced += 1
+            except Exception:
+                if not graceful:
+                    raise
+
+                _LOGGER.exception("Failed to sync reverse solver result with document id %r", document_id)
+                failed += 1
+        else:
+            _LOGGER.info("Sync of solver document with id %r skipped - already synced", document_id)
+            skipped += 1
```
Comment (pacospace) [inline thoth/storages/sync.py:197]: `revsolver`

File: thoth/storages/sync.py [hunk 3]
```diff
@@ -147,6 +148,57 @@ def sync_solver_documents(
...
+        processed += 1
+        if force or not graph.solver_document_id_exist(os.path.basename(document_id)):
+            try:
+                if is_local:
+                    _LOGGER.debug("Loading document from a local file: %r", document_id)
+                    with open(document_id, "r") as document_file:
+                        document = json.loads(document_file.read())
+                else:
+                    _LOGGER.info(
+                        "Syncing solver document from %r with id %r to graph", revsolver_store.ceph.host, document_id
```
Comment (pacospace) [inline thoth/storages/sync.py:183]: `revsolver`

File: thoth/storages/sync.py [hunk 4]
```diff
@@ -147,6 +148,57 @@ def sync_solver_documents(
...
+        graph.connect()
+
+    if not is_local:
+        revsolver_store = RevSolverResultsStore()
+        revsolver_store.connect()
+
+    processed, synced, skipped, failed = 0, 0, 0, 0
+    for document_id in document_ids or revsolver_store.get_document_listing():
+        processed += 1
+        if force or not graph.solver_document_id_exist(os.path.basename(document_id)):
```
Comment (pacospace) [inline thoth/storages/sync.py:175]: `revsolver`?

File: thoth/storages/sync.py (hunk 3 above)
Comment (fridex) [reply thoth/storages/sync.py]: Updated. :+1:

**Pull request merged**